src/League/ColorExtractor/Palette.php

```diff
@@ -68,7 +68,7 @@ public static function fromFilename($filename, $backgroundColor = null)
      */
     public static function fromGD($image, $backgroundColor = null)
     {
-        if (!is_resource($image) || get_resource_type($image) != 'gd') {
+        if (!$image instanceof \GDImage && (!is_resource($image) || 'gd' !== get_resource_type($image))) {
```

Comment:
```suggestion
        if (!($image instanceof \GDImage) && (!is_resource($image) || 'gd' !== get_resource_type($image))) {
```

Might need parenthesises here to avoid just evaluating `!$image` itself

Reply:
Nope `instanceof ` has precedence over ` ! ` (see https://www.php.net/manual/en/language.operators.precedence.php)

Reply:
Oh wow, I was not aware. Sorry about that. Learned something new about PHP today at least 😅